File: src/main/java/org/nanndeyanenw/loginreward/PlayerDataHandler.java
package org.nanndeyanenw.loginreward;

import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public class PlayerDataHandler {
        private File file;
        private FileConfiguration config;

        public PlayerDataHandler(File dataFolder, String fileName) {
            this.file = new File(dataFolder, fileName);
            if (!file.exists()) {
                try {
                    file.createNewFile();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            this.config = YamlConfiguration.loadConfiguration(file);
        }

        public void saveConfig() {
            try {
                config.save(file);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        public FileConfiguration getConfig() {
            return config;
        }
}

File: src/main/java/net/hotamachisubaru/loginreward/PlayerDataHandler.kt
package net.hotamachisubaru.loginreward

import org.bukkit.Bukkit
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File
import java.io.IOException

class PlayerDataHandler(dataFolder: File?, fileName: String?) {
    private val file: File
    @JvmField
    val config: FileConfiguration

    init {
        file = File(dataFolder, fileName)
        if (!file.exists()) {
            try {
                file.createNewFile()
            } catch (e: IOException) {
                Bukkit.getLogger().severe("ファイルを作成できませんでした。")
            }
        }
        config = YamlConfiguration.loadConfiguration(file)
    }

    fun saveConfig() {
        try {
            config.save(file)
        } catch (e: IOException) {
            Bukkit.getLogger().severe("設定ファイルを保存できませんでした。")
        }
    }
}